src/index.ts

```diff
@@ -282,3 +282,59 @@ export const gt = (versionA: string | null, versionB: string | null): boolean =>
 export const lt = (versionA: string | null, versionB: string | null): boolean => {
 	return compare(versionA, versionB) < 0;
 };
+
+/**
+ * @summary Check if a version satisfies a range
+ * @name satisfies
+ * @public
+ * @function
+ *
+ * @description Return true if the version satisfies the range
+ *
+ * @param {string|null} version - The version to evaluate
+ * @param {string} range - A semver range string, see the [node-semver](https://github.com/npm/node-semver#ranges)
+ * docs for details
+ *
+ * @returns {boolean} - True if the version satisfies the range, false otherwise
+ *
+ */
+export const satisfies = (version: string | null, range: string) => {
+	if (!version) {
+		return false;
+	}
+
+	version = normalize(<string>version);
+
+	if (!semver.valid(version)) {
+		return false;
+	}
+
+	return semver.satisfies(version, range);
+};
+
+/**
+ * @summary Return the lowest version in the list that satisfies the range
+ * @name maxSatisfying
+ * @public
+ * @function
+ *
+ * @description Return the highest version in the list that satisfies the range, or null if none of them do.
+ *
+ * @param {Array.<string|null>} versions - An array of versions to evaluate
+ * @param {string} range - A semver range string, see the [node-semver](https://github.com/npm/node-semver#ranges)
+ * docs for details
+ *
+ * @returns {string|null} - The highest matching version string, or null.
+ *
+ */
+export const maxSatisfying = (versions: Array<string | null>, range: string) => {
+	let max: string | null = null;
+
+	versions.forEach((version) => {
+		if (satisfies(version, range) && gt(version, max)) {
```

Comment:
> My only heads-up is that the satisfies in here will run normalize & semver.valid N times.

Personally I'm not super worried by this - we're unlikely to be checking maxSatisfying in any kind of performance critical situations (the download modal taking an extra 1ms to load is totally fine), and I think it'll make the overall code a meaningful amount worse: it's nice to have this directly depend on `satisfies`, because that guarantees that we get the same semantics. Doesn't make a big difference, and if people have strong opinions let's go for it, but by default I'd prefer nice code to microoptimized code, if it's not critical.

> Getting with it even further, someone might think that maxSatisfying can be the base plural/composite implementation of satisfies. In other words:
> ```
> satisfies(version, range) = maxSatisfying([version], range) != null;
> ```

Ah, interesting, you're saying that we could implement this like that? That would solve some of my concerns, by tying the two together. Sounds like it might make the overall code harder to follow, but I'm not sure - up to you @LucianBuzzo